js/test/Exchange/test.tradingFee.js

```diff
@@ -0,0 +1,28 @@
+'use strict'
+
+// ----------------------------------------------------------------------------
+
+const assert = require ('assert')
+
+//  ---------------------------------------------------------------------------
+
+function testTradingFee (exchange = undefined, symbol, fee) {
+    assert (fee, 'fee is undefined')
+    assert (fee['info'], 'fee info is empty')
+    assert (fee['symbol'] === symbol, 'trade symbol is not equal to requested symbol: trade: ' + fee['symbol'] + ' requested: ' + symbol)
+    if (fee['maker']) {
+        assert (typeof fee['maker'] === 'number', 'maker fee is not a number')
+    }
+    if (fee['taker']) {
+        assert (typeof fee['taker'] === 'number', 'taker fee is not a number')
+    }
+    if (fee['percentage']) {
```

Comment:
This is not entirely, if fee['percentage'] is `false`, then the test will be skipped.
So the test for a boolean will be skipped for all non-percentage fees.
We should rewrite this test a bit taking these rules into account: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#derived-exchange-classes (see how the other tests do that)

Reply:
Good catch!